docs: revamp landing page (#353)
Substra/substra-documentation: docs/source/index.rst

Substra documentation
=====================

> Substra is an open source federated learning (FL) software. It provides a flexible Python library and a web application to run federated learning training at scale.

Reviewer: "Software" is usually uncountable so this sentence is a bit weird. I usually say it is a federated learning framework, is that bad?

Author: we decided to use the word software and not framework a year ago. Let's discuss this more in person.

.. image:: banner.jpg
  :width: 800
  :align: center
  :alt: Substra

Substra's main usage is in production environments. It has already been deployed and used by hospitals and biotech companies: see the `MELLODDY <https://www.melloddy.eu/>`_ and the `HealthChain <https://www.nature.com/articles/s41591-022-02155-w>`_ projects.

The key Substra differentiators are:

* **Framework agnostic** — Any Python library can be used: PyTorch, TensorFlow, sklearn, etc.
* **Flexible** — Any kind of computation can be run: machine learning, analytics, etc.
* **Scalable** — Support for vertical scaling (several trainings on one machine) and horizontal scaling (training on several machines).

> * **Traceable** — All machine learning actions are traced.

Author: ```suggestion
* **Traceable** — All machine learning operations are logged in an auditable read-only database.
```

Substra documentation
=====================

Substra is an open source federated learning (FL) software. It provides a flexible Python library and a web application to run federated learning training at scale.

Substra's main usage is in production environments. It has already been deployed and used by hospitals and biotech companies: see the `MELLODDY <https://www.melloddy.eu/>`_ and the `HealthChain <https://www.nature.com/articles/s41591-022-02155-w>`_ projects.

.. image:: banner.jpg
  :width: 600
  :align: center
  :alt: Substra

The key Substra differentiators are:

* **Framework agnostic** Any Python library can be used: PyTorch, TensorFlow, sklearn, etc.
* **Flexible** Any kind of computation can be run: machine learning, analytics, etc.
* **Scalable** Support for vertical scaling (several trainings on one machine) and horizontal scaling (training on several machines).

> * **Traceable** All machine learning actions are traced.

Reviewer: I think it might be bad to use "trace" in both the word and its description, maybe we can say something like this (just a suggestion):

```suggestion
* **Traceable** All machine learning operations are logged in an auditable read-only store.```

Author: Thanks for the proposition.

I think it's better. But I find the word `store` weird? Do you have a more understandable / general purpose word? Database maybe?

Reviewer: 🤔 I suppose a database is a specific implementation of a store but I think database is appropriate.

* **Production ready** Packaged in Kubernetes and regularly audited.

> * **Debugging made easy** Remote error logs accessible for data-scientist. The same code can be run in a deployed production environment or on a single machine to debug.

Reviewer: ```suggestion
* **Debugging made easy** Remote error logs are accessible to data scientists. The same code can be run in a deployed production environment or on a single machine to debug.
```

Reviewer: Also don't let Jean or Sarah see "debugging made easy" as a Substra selling point >_>
